src/services/transcript-parser.js

```diff
@@ -827,37 +832,55 @@ export const getMatchedTranscriptLines = (searchHits, query, transcripts) => {
     }
     const matchOffset = mappedText.toLocaleLowerCase().indexOf(qStr);
     if (matchOffset !== -1 && transcirptId != undefined) {
-      const matchParts = getMatchedParts(matchOffset, mappedText, qStr);
+      const match = markMatchedParts(mappedText, qStr);
 
       transcriptLines.push({
         ...hit,
         begin: start,
         end: end,
         id: transcirptId,
-        match: matchParts,
+        match,
+        matchCount: item.hitCount,
         text: value,
       });
     }
   });
   return transcriptLines;
 };
 
-// FIXME:: When there are 2 hits in the same transcript text/cue, only the first
-// match is highlighted.
 /**
- * Generic function to split the matched transcript text into 3 parts where the output is in
- * the format [text before search query, search query, text after search query]
- * @param {Number} offset character offset to the query string in the matched transcript text/cue
+ * Generic function to mark the matched transcript text in the cue where the output has
+ * <span class="ramp--transcript_highlight"></span> surrounding the matched parts
+ * within the cue.
  * @param {String} text matched transcript text/cue
  * @param {String} query current search query
- * @returns a list of parts of the given matched transcript text/cue
+ * @returns matched cue with HTML tags added for marking the hightlight 
  */
-export const getMatchedParts = (offset, text, query) => {
-  return [
-    text.slice(0, offset),
-    text.slice(offset, offset + query.length),
-    text.slice(offset + query.length)
-  ];
+export const markMatchedParts = (text, query) => {
+  const queryRegex = new RegExp(String.raw`${query}`, 'gi');
```

Comment:
This appears to be too broad a regex.  When I search for 'you' on the first section of https://avalon-dev.dlib.indiana.edu/media_objects/g158bh28p/manifest.json, I get 35 hits but 55 highlights.  From looking at the content search results it looks like 35 is correct.  Some examples of false highlights include `your`, `you'd` and `you'll`).

Reply:
Good catch! I'll take a look at it.